integration/simulation/simulation_docker_test.go

```diff
@@ -0,0 +1,117 @@
+package simulation
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+)
+
+// TODO - Use individual Docker containers for the Obscuro nodes and Ethereum nodes.
+
+var (
+	enclaveDockerImg  = "obscuro_enclave"
+	nodeIDFlag        = "--nodeID"
+	enclaveDockerPort = "11000/tcp"
+)
+
+// This test creates a network of L2 nodes, then injects transactions, and finally checks the resulting output blockchain
+// The L2 nodes communicate with each other via sockets, and with their enclave servers via RPC.
+// All nodes live in the same process, the enclaves run in individual Docker containers, and the Ethereum nodes are mocked out.
+func TestDockerNodesMonteCarloSimulation(t *testing.T) {
+	logFile := setupTestLog()
+	defer logFile.Close()
+
+	params := SimParams{
+		NumberOfNodes:         10,
+		NumberOfWallets:       5,
+		AvgBlockDurationUSecs: uint64(250_000),
+		SimulationTimeSecs:    15,
+	}
+	params.AvgNetworkLatency = params.AvgBlockDurationUSecs / 15
+	params.AvgGossipPeriod = params.AvgBlockDurationUSecs / 3
+	params.SimulationTimeUSecs = params.SimulationTimeSecs * 1000 * 1000
+
+	efficiencies := EfficiencyThresholds{0.2, 0.3, 0.4}
+
+	ctx := context.Background()
```

Comment:
can you add some comments for each step of this test?
For people who are not very familiar with the docker stuff ( like myself)

Reply:
Will add, thanks.